app/models/alert.rb

```diff
@@ -67,8 +67,7 @@ def email_course_admins
   def email_target_user
     return if emails_disabled?
     return if target_user.nil?
-    AlertMailer.alert(self, target_user).deliver_now
-    update_attribute(:email_sent_at, Time.now)
+    AlertMailerWorker.schedule_email(alert: self, user: target_user)
```

Comment:
The place we should intervene is the controller, rather than the model.

In some cases, there are batches of alerts generated and sent, and we throttle those email sending events to make gmail happy. The only reason we want to do them async is to avoid the specific case of sending an email within an http request.

The place to call the worker would be `AlertsController#email_target_user` (passing just the `alert.id` to `perform_async`)